### PR TITLE
[Communication] Update to swagger version 2020-08-20

### PR DESCRIPTION
**Pull request by memontic-ms**
## Checklist

- [x] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [ ] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [ ] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [ ] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
        - {Please put the link here}
    - [ ] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

File: src/Communication/examples/Test-AzCommunicationServiceNameAvailability.md [hunk 1]
```diff
@@ -0,0 +1,18 @@
+### Example 1: {{ Add title here }}
```
Comment (dolauli) [inline src/Communication/examples/Test-AzCommunicationServiceNameAvailability.md:1]: You need to populate these examples.

File: src/Communication/test/Test-AzCommunicationServiceNameAvailability.Tests.ps1 [hunk 2]
```diff
@@ -0,0 +1,30 @@
+$loadEnvPath = Join-Path $PSScriptRoot 'loadEnv.ps1'
```
Comment (dolauli) [inline src/Communication/test/Test-AzCommunicationServiceNameAvailability.Tests.ps1:1]: You need to implement tests for this cmdlet based on the generated test stubs.

**Review by dolauli — CHANGES_REQUESTED** [review]
Generally, you need to update test/examples, and re-record the test.

**Comment by BethanyZhou** [comment]
Gently ping~

**Comment by isra-fel** [comment]
Closing the PR for lack of activity. Feel free to reopen to continue working on it. Thanks.

**Comment by memontic-ms** [comment]
Reopening.

**Review by isra-fel — CHANGES_REQUESTED** [review]
Hi @memontic-ms 

- Pretty much only the default parameter set is usable. We might want to hide (remove) the rest ([how to](https://github.com/Azure/autorest.powershell/blob/main/docs/directives.md#cmdlet-removal). Note that a "variant" means a powershell parameter set.)
- Is `-Type` always going to be "Microsoft.Communication/CommunicationServices"? If so, we should considering hiding this parameter and use a default hardcoded value instead ([how to](https://github.com/Azure/autorest.powershell/blob/main/docs/directives.md#Parameter-Hiding)); or if it's just most case, we should consider making it not manadatory by adding a default value ([how to](https://github.com/Azure/autorest.powershell/blob/main/docs/directives.md#default-values)). Either saves the effort for our customer to input a long constant string.

**Comment by memontic-ms** [comment]
> Hi @memontic-ms
> 
> * Pretty much only the default parameter set is usable. We might want to hide (remove) the rest ([how to](https://github.com/Azure/autorest.powershell/blob/main/docs/directives.md#cmdlet-removal). Note that a "varient" means a powershell parameter set.)
> * Is `-Type` always going to be "Microsoft.Communication/CommunicationServices"? If so, we should considering hiding this parameter and use a default hardcoded value instead ([how to](https://github.com/Azure/autorest.powershell/blob/main/docs/directives.md#Parameter-Hiding)); or if it's just most case, we should consider making it not manadatory by adding a default value ([how to](https://github.com/Azure/autorest.powershell/blob/main/docs/directives.md#default-values)). Either saves the effort for our customer to input a long constant string.

I have set a default value for the -Type parameter of Test-AzCommunicationServiceNameAvailability and pushed those changes to this PR.

I don't know what you mean by the first thing. Do you want me to remove the variations of Test-AzCommunicationServiceNameAvailability that have NameAvailabilityParameter? I tried to do that but couldn't get it to work. 
Is the following what is needed? It doesn't seem to remove those variants.
```
  - where:
      verb: Test
      variant: ^NameAvailabilityParameter$
      subject: CommunicationServiceNameAvailability
    remove: true
```

File: src/Communication/docs/Test-AzCommunicationServiceNameAvailability.md [hunk 3]
```diff
@@ -0,0 +1,227 @@
...
+Test-AzCommunicationServiceNameAvailability -InputObject <ICommunicationIdentity>
+ -NameAvailabilityParameter <INameAvailabilityParameters> [-DefaultProfile <PSObject>] [-Confirm] [-WhatIf]
+ [<CommonParameters>]
+```
+
+### CheckViaIdentityExpanded
+```
+Test-AzCommunicationServiceNameAvailability -InputObject <ICommunicationIdentity> -Name <String>
+ [-Type <String>] [-DefaultProfile <PSObject>] [-Confirm] [-WhatIf] [<CommonParameters>]
+```
```
Comment (isra-fel) [inline src/Communication/docs/Test-AzCommunicationServiceNameAvailability.md:38]: Actually I want to remove these three parameter sets. They are generated for piping or other purpose but do not apply in the use case of this cmdlet.
Try this:
```yaml
  - where:
      verb: Test
      variant: ^Check$|^CheckViaIdentity$|^CheckViaIdentityExpanded$
      subject: CommunicationServiceNameAvailability
    remove: true
```

Comment (isra-fel) [reply src/Communication/docs/Test-AzCommunicationServiceNameAvailability.md:38]: Also please remember to run `build-module.ps1` and check in the updated help file

**Review by isra-fel — CHANGES_REQUESTED** [review]

File: src/Communication/docs/Test-AzCommunicationServiceNameAvailability.md (hunk 3 above)
Comment (memontic-ms) [reply src/Communication/docs/Test-AzCommunicationServiceNameAvailability.md]: Perfect, thanks. I've updated with those changes.

**Review by isra-fel — APPROVED** [review]
Looks good!

**Pull request merged**